test/badsalt.c

```diff
@@ -252,7 +252,8 @@ check_results (const char *label, const char *fn,
     {
       if (retval[0] == '*' ||
           strncmp (retval, setting,
-                   (setting[l_setting - 1] == '*') ? l_setting - 1 : l_setting))
+                   (setting[l_setting - 1] == '*' ||
+                    setting[l_setting - 1] == ':') ? l_setting - 1 : l_setting))
```

Comment:
What changed to make this necessary?

Reply:
The asterisk `*` character is a valid character inside of the salt for {md5,sha{256,512}} crypt hashes.